server/aesdsocket.c

```diff
@@ -0,0 +1,420 @@
+	}
+	syslog(LOG_INFO, "socket succeeded!\n");
+	#if DEBUG
+		printf("socket() succeeded\n");
+	#endif	
+	/*------------------------------------------------------------------------------------------------------------------------------------------------*/
+
+	//bind
+	status = bind(sockfd, servinfo->ai_addr, servinfo->ai_addrlen);
+	if (status == ERROR)
+	{
+		syslog(LOG_ERR, "Error Binding\n");
+		#if DEBUG
+			printf("bind() failed\n");
+		#endif
+		clear();
+		return ERROR;
+	}
+	syslog(LOG_INFO, "bind succeeded!\n");
+	#if DEBUG
+		printf("bind() succeeded\n");
+	#endif	
+	/*------------------------------------------------------------------------------------------------------------------------------------------------*/
+
+	//create a file
+	int fd = creat(STORAGE_PATH, FILE_PERMISSIONS);
+	if (fd == ERROR)
+	{
+		syslog(LOG_ERR, "File cannot be created");
+		#if DEBUG
+			printf("creat() failed\n");
+		#endif
+		clear();
+		return ERROR;
+	}
+	close(fd);
+	syslog(LOG_INFO, "File created!\n");
+	#if DEBUG
+		printf("creat() succeeded\n");
+	#endif	
+	/*------------------------------------------------------------------------------------------------------------------------------------------------*/
+
+	while (1)
+	{
+        buffer = (char *) malloc((sizeof(char)*BUFFER_SIZE));
+		if(buffer == NULL)
+		{
+			syslog(LOG_ERR, "Malloc failed!\n");
+			#if DEBUG
+				printf("malloc() failed\n");
+			#endif
+			clear();
+			return ERROR;
+		}
+		syslog(LOG_INFO, "malloc succeded!\n");
+		#if DEBUG
+			printf("malloc() succeded\n");
+		#endif
+		memset(buffer,0,BUFFER_SIZE);
+		/*------------------------------------------------------------------------------------------------------------------------------------------------*/
+
+		//listen
+		status = listen(sockfd, 5);
+		if(status == ERROR)
+		{
+			syslog(LOG_ERR, "Error Listening\n");
+			#if DEBUG
+				printf("listen() failed\n");
+			#endif
+			clear();
+			return ERROR;
+		}
+		syslog(LOG_INFO, "listen() succeeded!\n");
+		#if DEBUG
+			printf("listen() succeeded\n");
+		#endif
+		/*------------------------------------------------------------------------------------------------------------------------------------------------*/
+
+		//accept
+		clientfd = accept (sockfd, (struct sockaddr *)&clientaddr, &sockaddrsize); //check here
+		if (clientfd == ERROR)
+		{
+			syslog(LOG_ERR, "Error accepting\n");
+			#if DEBUG
+				printf("accept() failed\n");
+			#endif
+			clear();
+			return ERROR;
+		}
+		syslog(LOG_INFO, "accept succeeded! Accepted connection from: %s\n", clientaddr.sa_data);
+		#if DEBUG
+			printf("accept succeeded! Accepted connection from: %s\n",clientaddr.sa_data);
+		#endif
+		/*------------------------------------------------------------------------------------------------------------------------------------------------*/
+
+		bool packet_rcvd = false;
+		int i;
+		status = 0;
+
+		while(packet_rcvd == false)
+		{
+			nread = recv(clientfd, read_data, BUFFER_SIZE, 0);
```

Comment:
what will happen if the packet is more than 1024 bytes before you receive your first '\n'? also does this comply with the requirement as stated in https://www.coursera.org/learn/advanced-embedded-software-development/supplement/N9ZTJ/assignment-5-part-1-instructions which says that the file should be written after newline? This implementation would not work for assignment 8